outline/connectivity/connectivity.go

```diff
@@ -32,16 +33,64 @@ const (
 	bufferLength        = 512
 )
 
-// AuthenticationError is used to signal failed authentication to the Shadowsocks proxy.
-type AuthenticationError struct {
+// Outline error codes. Must be kept in sync with definitions in outline-client/cordova-plugin-outline/outlinePlugin.js
+const (
+	NoError                     = 0
+	Unexpected                  = 1
+	NoVPNPermissions            = 2
+	AuthenticationFailure       = 3
+	UDPConnectivity             = 4
+	Unreachable                 = 5
+	VpnStartFailure             = 6
+	IllegalConfiguration        = 7
+	ShadowsocksStartFailure     = 8
+	ConfigureSystemProxyFailure = 9
+	NoAdminPermissions          = 10
```

Comment:
I feel it is a little bit weird to put these error codes under `connectivity`, because some of them (such as `NoAdminPermissions` and `IllegalConfiguration`) are not related to connectivity tests at all. Maybe we can put it under `outline/error_code.go`?

Reply:
It does. But it reflects the fact that the functions are not defining their own API, and instead we are borrowing an API from outline-client. We need to revamp our error handling, but that's out of scope of this change.

For now, I introduced a `neterrors` package to hold the shared error definition, so the other packages can borrow from it. It's really only the connectivity package and the Electron main that need the definitions.